app/pages/_app.tsx

```diff
@@ -20,6 +21,9 @@ const initialPreloadedQuery = getInitialPreloadedQuery({
 });
 
 function MyApp({ Component, pageProps }: AppProps) {
+  const [error, setError] = useState(null);
+  const value = { error, setError };
+
```

Comment:
you can have a `useEffect` resetting the error to `null` when `Component` changes